btree/btree.c

```diff
@@ -2900,27 +2941,47 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 			return fail(bop, oi->i_nop.no_op.o_sm.sm_rc);
 		}
 	case P_ALLOC: {
-		int ksize;
-		int vsize;
-		/* Validate curr_level->l_node(i.e.is it still exists or not) */
-		int rc = node_is_valid(curr_level->l_node);
-		if (rc) {
-			node_op_fini(&oi->i_nop);
-			return fail(bop, rc);
-		}
-		if (curr_level->l_node->n_type->nt_id == BNT_FIXED_FORMAT) {
-			ksize = node_keysize(curr_level->l_node);
-			vsize = node_valsize(curr_level->l_node);
-		} else {
-			/**
-			 * @todo : ksize, vsize for variable key:
-			 * get the correct max key size at runtime
-			 * */
-			ksize = MAX_KEY_SIZE;
-			vsize = MAX_VAL_SIZE;
-		}
-		if (m0_btree_overflow_is_possible(curr_level->l_node, ksize,
-						  vsize))
+		struct level *lev;
+		int           ksize;
+		int           vsize;
+		bool          load = false;
+		int           rc;
+		do {
+			lev = &oi->i_level[oi->i_used];
+			/* Validate curr_level->l_node */
+			rc = node_is_valid(lev->l_node);
+			if (rc) {
+				node_op_fini(&oi->i_nop);
+				return fail(bop, rc);
+			}
+			if (lev->l_node->n_type->nt_id == BNT_FIXED_FORMAT) {
+				ksize = node_keysize(lev->l_node);
+				vsize = node_valsize(lev->l_node);
+			} else {
+				/**
+				 * @todo : ksize, vsize for variable key:
+				 * get the correct max key size at runtime
+				 * */
+				ksize = MAX_KEY_SIZE;
+				vsize = MAX_VAL_SIZE;
+			}
```

Comment:
Why don't we have 
ksize = m0_vec_count(&bop->bo_rec.r_key.k_data.ov_vec)
vsize = m0_vec_count(&bop->bo_rec.r_val.ov_vec)

Reply:
It will be work for fixed size format not for variable format , Example, for leaf node we can use m0_vec_count(&bop->bo_rec.r_key.k_data.ov_vec), m0_vec_count(&bop->bo_rec.r_val.ov_vec) to determine overflow condition.but for internal node key size will be different